### PR TITLE
docs: Bigtable Load generator tool

### PR DESCRIPTION
**Pull request by billyjacobson**
A dataflow template to put a heavy read load on a Bigtable table for demonstration purposes

> It's a good idea to open an issue first for discussion.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: bigtable/beam/workload-generator/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+### Deploying instructions
+
+1. Build the project
+
+    ```
+    mvn clean package -DskipTests
+    ```
+
+1. Set the environment variables. To deploy a version on your project, update
```
[the remark this answers is not in the document]
Comment (kolea2) [reply bigtable/beam/workload-generator/README.md:47]: Got it, thanks! Mind adding some more details from this comment into the README?

File: bigtable/beam/workload-generator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    <dependency>
+      <groupId>com.google.cloud.bigtable</groupId>
+      <artifactId>bigtable-hbase-beam</artifactId>
+      <version>1.25.1</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-bigtable</artifactId>
```
[the remark this answers is not in the document]
Comment (billyjacobson) [reply bigtable/beam/workload-generator/pom.xml:112]: I typicaly use the veneer for tests cus it's easier, but I don't seem to have done that in this case. I will either remove or use it in the tests and add a test scope

File: bigtable/beam/workload-generator/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# Bigtable workload generator
+
+This is a tool to perform a high number of reads to a Bigtable table for
+demonstration purposes. It is deployed as a Dataflow template, so it can easily
+be run.
```
Comment (kolea2) [inline bigtable/beam/workload-generator/README.md:5]:
```suggestion
be run as a Dataflow job.
```

File: bigtable/beam/workload-generator/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+### Running
+
+1. Set your environment variables
+
+    ```
+    TEMPLATE_PATH="gs://cloud-bigtable-dataflow-templates/generate-workload.json"
+    INSTANCE_ID=YOUR-INSTANCE-ID
+    TABLE_ID=YOUR-TABLE-ID
+    WORKLOAD_QPS=100 # Optional
```
Comment (kolea2) [inline bigtable/beam/workload-generator/README.md:17]: nit: maybe move the optional params last?

File: bigtable/beam/workload-generator/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    --region "$REGION"
+    ```
+
+1. Make sure to cancel the job once you are done.
+
+    ```
+    gcloud dataflow jobs cancel $JOB_NAME
+    ```
+
+### Deploying instructions
```
Comment (kolea2) [inline bigtable/beam/workload-generator/README.md:39]:
```suggestion
### Deploying a template instructions
```

File: bigtable/beam/workload-generator/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+If you would like to modify this and run it yourself you can use these commands:
+
+1. Create a Bigtable instance and table
+
+1. Set up the environment variables
+
+   ```
+   GOOGLE_CLOUD_PROJECT=your-project-id
+   INSTANCE_ID=your-instance-id
+   TABLE_ID=your-table-id WORKLOAD_QPS=100 # Optional
```
Comment (kolea2) [inline bigtable/beam/workload-generator/README.md:83]: same suggestion here, maybe move `WORKLOAD_QPS` to last

File: bigtable/beam/workload-generator/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+   REGION=us-central1
+   ```
+
+1. Run the command
+
+   ```
+   mvn compile exec:java -Dexec.mainClass=WorkloadGenerator /
+   "-Dexec.args=--bigtableInstanceId=$INSTANCE_ID =--bigtableTableId=$TABLE_ID /
+   --runner=dataflow --project=$GOOGLE_CLOUD_PROJECT /
+   --workloadQPS=$WORKLOAD_QPS --region=$REGION"
```
Comment (kolea2) [inline bigtable/beam/workload-generator/README.md:93]: here too

File: bigtable/beam/workload-generator/pom.xml [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>31.0.1-jre</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.cloud.bigtable</groupId>
+      <artifactId>bigtable-hbase-beam</artifactId>
+      <version>1.25.1</version>
```
Comment (kolea2) [inline bigtable/beam/workload-generator/pom.xml:107]: this is now 1.26.1 :)

File: bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+      descriptor.addFamily(new HColumnDescriptor(COLUMN_FAMILY_NAME));
+      admin.createTable(descriptor);
+    } catch (Exception e) {
+      System.out.println("Error during beforeClass: \n" + e.toString());
+    }
+  }
+
+  @Before
+  public void setupStream() {
+    // bout = new ByteArrayOutputStream();
```
Comment (kolea2) [inline bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java:95]: do we need this?

File: bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+    long startRequestCount = 0;
+    long endRequestCount = 0;
+    for (TimeSeries ts : response.iterateAll()) {
+      startRequestCount = ts.getPoints(0).getValue().getInt64Value();
+      endRequestCount = ts.getPoints(ts.getPointsCount() - 1).getValue().getInt64Value();
+    }
+    assertThat(endRequestCount - startRequestCount > rate);
+
+    // Stop the running job.
+    String jobId = ((DataflowPipelineJob) pipelineResult).getJobId();
```
Comment (kolea2) [inline bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java:184]: should this be in a try/catch/finally so we always stop the job? or would the job stop itself if there are errors?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java (hunk 10 above)
Comment (billyjacobson) [reply bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java:184]: if there is an issue the job would probably fail to build

File: bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java (hunk 9 above)
Comment (billyjacobson) [reply bigtable/beam/workload-generator/src/test/java/bigtable/WorkloadGeneratorTest.java:95]: yes, I uncommented for debugging, will re add

File: bigtable/beam/workload-generator/README.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+1. Create a Bigtable instance and table
+
+1. Set up the environment variables
+
+   ```
+   GOOGLE_CLOUD_PROJECT=your-project-id
+   INSTANCE_ID=your-instance-id
+   REGION=us-central1
+   TABLE_ID=your-table-id WORKLOAD_QPS=100 # Optional
```
Comment (kolea2) [inline bigtable/beam/workload-generator/README.md:88]: nit - I think `WORKLOAD_QPS` needs to be on a new line

**Review by kolea2 — APPROVED** [review]
LGTM after one small change to the readme. Thank you!

**Pull request merged**